plugins/inventory/vmware_vm_inventory.py

```diff
@@ -247,88 +336,116 @@ def check_requirements(self):
+        if len(containers) == 0:
+            return []
+
+        objs_list = [ObjectSpec(
+            obj=self.content.viewManager.CreateContainerView(r, [vim_type], True),
+            selectSet=[TraversalSpec(path='view', skip=False, type=vim.view.ContainerView)]) for r in containers]
+
+        is_all = False if properties else True
 
         # Create Property Spec
-        property_spec = vmodl.query.PropertyCollector.PropertySpec(
+        property_spec = PropertySpec(
             type=vim_type,  # Type of object to retrieved
-            all=False,
+            all=is_all,
             pathSet=properties
         )
 
-        # Create Object Spec
-        object_spec = vmodl.query.PropertyCollector.ObjectSpec(
-            obj=mor,
-            skip=True,
-            selectSet=[traversal_spec]
-        )
-
         # Create Filter Spec
-        filter_spec = vmodl.query.PropertyCollector.FilterSpec(
-            objectSet=[object_spec],
+        filter_spec = FilterSpec(
+            objectSet=objs_list,
             propSet=[property_spec],
             reportMissingObjectsInResults=False
         )
 
-        return self.content.propertyCollector.RetrieveContents([filter_spec])
-
-    @staticmethod
-    def _process_object_types(vobj, level=0):
-        """For an object that is not a valid JSON type, loop over its properties
-        and return them in a dictionary"""
         try:
-            json.dumps(vobj)
-            return vobj
-        except Exception:
-            rdata = {}
-            properties = dir(vobj)
-            properties = [str(x) for x in properties if not x.startswith('_')]
-            properties = [x for x in properties if x not in ('Array', 'disabledMethod', 'declaredAlarmState')]
-            properties = sorted(properties)
-
-            for prop in properties:
-                # Attempt to get the property, skip on fail
-                try:
-                    propToSerialize = getattr(vobj, prop)
-                except Exception:
-                    continue
-
-                if callable(propToSerialize):
-                    continue
-
-                prop = prop.lower()
-                if level + 1 <= 2:
-                    try:
-                        rdata[prop] = BaseVMwareInventory._process_object_types(
-                            propToSerialize,
-                            level=(level + 1)
-                        )
-                    except vim.fault.NoPermission:
-                        pass
-
-        return rdata
+            return self.content.propertyCollector.RetrieveContents([filter_spec])
+        except vmodl.query.InvalidProperty as err:
+            _handle_error("Invalid property name: %s" % err.name)
+        except Exception as err:  # pylint: disable=broad-except
+            _handle_error("Couldn't retrieve contents from host: %s" % to_native(err))
+        return []
 
     @staticmethod
     def _get_object_prop(vm, attributes):
```

Comment:
This method is no longer reference by anything, because the 1 line that calls it was deleted.

So this is from me tracking the source of top-level variables going missing. I haven't connected the dots, but it seems particular unsafe to stop using the logic here, because it means the hostvars schema is changed.